subsys/mgmt/ec_host_cmd/backends/ec_host_cmd_backend_uart.c

```diff
@@ -0,0 +1,303 @@
+	 */
+	USART_HOST_CMD_STATE_DISABLED,
+
+	/*
+	 * Ready to receive next request
+	 * This state represents USART layer is initialized and ready to
+	 * receive host request. Once the response is sent, current_state is
+	 * reset to this state to accept next packet.
+	 */
+	USART_HOST_CMD_READY_TO_RX,
+
+	/*
+	 * Receiving request
+	 * After first byte is received current_state is moved to receiving
+	 * state until all the header bytes + datalen bytes are received.
+	 * If host_request_timeout was called in this state, it would be
+	 * because of an underrun situation.
+	 */
+	USART_HOST_CMD_RECEIVING,
+
+	/*
+	 * Processing request
+	 * Once the process_request starts processing usart_in_buffer,
+	 * current_state is moved to processing state. Host should not send
+	 * any bytes in this state as it would be considered contiguous
+	 * request.
+	 */
+	USART_HOST_CMD_PROCESSING,
+
+	/*
+	 * Sending response
+	 * Once host task is ready with the response bytes, current_state is
+	 * moved to sending state.
+	 */
+	USART_HOST_CMD_SENDING,
+
+	/*
+	 * Received bad data
+	 * If bad packet header is received, current_state is moved to rx_bad
+	 * state and after rx_timeout all the bytes are dropped.
+	 */
+	USART_HOST_CMD_RX_BAD,
+
+	/*
+	 * Receiving data overrun bytes
+	 * If extra bytes are received after current_state is in complete,
+	 * host is sending extra bytes which indicates data overrun.
+	 */
+	USART_HOST_CMD_RX_OVERRUN,
+};
+
+struct ec_host_cmd_uart_ctx {
+	/* uart device instance */
+	const struct device *uart_dev;
+	/* Context for read and tx operations */
+	struct ec_host_cmd_rx_ctx *rx_ctx;
+	struct ec_host_cmd_tx_buf *tx;
+	struct k_work_delayable timeout_work;
+	enum uart_host_command_state state;
+	uint32_t expected_len;
+};
+
+static int request_expected_size(const struct ec_host_cmd_request_header *r)
+{
+	/* Check host request version */
+	if (r->prtcl_ver != 3) {
+		return 0;
+	}
+
+	/* Reserved byte should be 0 */
+	if (r->reserved) {
+		return 0;
+	}
+
+	return sizeof(*r) + r->data_len;
+}
+
+#define EC_HOST_CMD_UART_DEFINE(_name)                                                             \
+	static struct ec_host_cmd_uart_ctx _name##_hc_uart;                                        \
+	struct ec_host_cmd_backend _name = {                                                       \
+		.api = &ec_host_cmd_api,                                                           \
+		.ctx = (struct ec_host_cmd_uart_ctx *)&_name##_hc_uart,                            \
+	}
+
+/* Waiting time in microseconds to detect overrun */
+#define OVERRUN_TIMEOUT	     300
+/* Timeout after receiving first byte */
+#define USART_REQ_RX_TIMEOUT K_MSEC(150)
+
+static void rx_timeout(struct k_work *work)
+{
+	struct k_work_delayable *dwork = k_work_delayable_from_work(work);
+	struct ec_host_cmd_uart_ctx *hc_uart =
+		CONTAINER_OF(dwork, struct ec_host_cmd_uart_ctx, timeout_work);
+	int res;
+
+	LOG_ERR("USART HOST CMD ERROR: ");
```

Comment:
LOG* functions have an implicit `\n`, this would look weird on the console. Drop this one and drop the '\n' from the ones below, that'd be good enough.

Reply:
Done.

Reply:
Ok but either drop this one now or at least remove the trailing space.

Reply:
Done.